add text-right tag to TableHelper.

imo-tikuwa opened the pull request:
#15117

imo-tikuwa committed:
add text-right tag to TableHelper.

dereuromark commented:
I think you should point a new feature such as this to 4.next branch (as master is feature frozen, only bugfixes).

dereuromark commented on src/Shell/Helper/TableHelper.php:
```diff
@@ -114,7 +114,11 @@ protected function _render(array $row, array $widths, array $options = []): void
                 $column = $this->_addStyle($column, $options['style']);
             }
-            $out .= '| ' . $column . str_repeat(' ', $pad) . ' ';
+            if (!is_null($column) && preg_match('#<text-right>.*</text-right>#', $column)) {
```
```suggestion
            if ($column !== null && preg_match('#<text-right>.+</text-right>#', $column)) {
```

markstory replied:
What will happen when someone does `<text-right>some</text-right>text`?

imo-tikuwa replied:
Thank you for pointing out the code.
I took in the points that you pointed out.

When I tried `<text-right>some</ text-right>text`, it was right-justified, including the text outside the tag.
```
$outputs = [
    ['Item', 'Price per piece (yen)'],
    ['Apple', '<text-right>some</text-right>text'],
    ['Orange', '100'],
];
$io->helper('Table')->output($outputs);
```
![image](https://user-images.githubusercontent.com/48991931/97931131-4f7d6680-1db0-11eb-83f3-3420fd35bdab.png)

markstory replied:
I assumed that would be the output. I don't think it is expected based on the markup text though. The CLI markup language borrows conventions from HTML which would generate results like `text      some` instead. We could raise an exception if there is text outside the alignment tags as it is likely an error.

It would also be good to ensure formatting like `<text-right><comment>blue</comment>normal</text-right>` behaves correctly as well.

imo-tikuwa committed:
Update src/Shell/Helper/TableHelper.php

Co-authored-by: Mark Sch. <dereuromark@users.noreply.github.com>

dereuromark commented:
> What will happen when someone does <text-right>some</text-right>text?

Yeah, I think we should have a test case for this.
It could through an exception maybe.

imo-tikuwa committed:
Added exception handling, etc

Exclude the text-right tag at the timing before passing it to the output processing of the console
Supports processing of nested color tags

imo-tikuwa commented:
@markstory 
@dereuromark 
I modified it to throw an exception when there is a string outside the text-right tag.

> It would also be good to ensure formatting like `<text-right><comment>blue</comment>normal</text-right>` behaves correctly as well.

I tried to correspond to tags such as info included in the text-right tag, the info tag was output as a character string.
`<text-right>` and `</ text-right>` are used to determine if `str_repeat(' ', $pad)` in the TableHeler class should be done before or after the string.
Therefore, I thought that it would be okay to delete it from the character string at the timing before passing it to the output processing of the console, so I fixed it.
